### PR TITLE
vdk-plugins: improve plugin project creation with cookiecutter

### PR DESCRIPTION
**Pull request by tozka**
Add quickstart to make it easier to create a new plugin project with
cookiecutter instead of copy pasting the template folder (which was not
used a lot)

Now it is as easy as

```
pip install -U cookiecutter
cookiecutter https://github.com/tozka/cookiecutter-vdk-plugin.git
```

Testing Done: 
`cookiecutter https://github.com/tozka/cookiecutter-vdk-plugin.git`

Signed-off-by: Antoni Ivanov <aivanov@vmware.com>

### CODE REVIEW

**Review by ivakoleva — CHANGES_REQUESTED** [review]

File: projects/vdk-plugins/README.md [hunk 1]
```diff
@@ -21,6 +21,27 @@ Once the plugin is installed, vdk automatically finds it and activates it.
...
+Install the latest Cookiecutter if you haven't installed it yet (this requires Cookiecutter 1.4.0 or higher):
+
+```
+pip install -U cookiecutter
+```
+
+Generate a VDK Plugin package project:
+
+```
+cookiecutter https://github.com/tozkata/cookiecutter-vdk-plugin.git
```
[the remark this answers is not in the document]
Comment (tozka) [reply projects/vdk-plugins/README.md:35]: This is the template. It needs to be separate repo, And yes, it's public - https://github.com/tozka/cookiecutter-vdk-plugin.git

[the remark this answers is not in the document]
Comment (ivakoleva) [reply projects/vdk-plugins/README.md:35]: Maybe add a sentence about the template repo in the README.md, and the other one you added about plugin code generation?
Will try it and provide feedback; Should we add smoke test coverage?

[the remark this answers is not in the document]
Comment (tozka) [reply projects/vdk-plugins/README.md:35]: I don't undertstand about the comment, @ivakoleva 

The template repo is in the README already: in the command: 

Smoke test coverage - probably not now.

**Review by gageorgiev — APPROVED** [review]
Looks good. One thing that worries me is whether it's appropriate to have the template in your personal projects. Could we not rework the current plugin template to be a submodule (https://git.wiki.kernel.org/index.php/GitSubmoduleTutorial) and pass a path to it when creating a plugin?

**Pull request merged**